File: Algorithm/binary_search.py
import math

def binary_search(list, target):
    left = 0
    right = len(list) - 1
    
    while left < right:
        middle = math.floor((left + right) / 2)
        
        if target <= list[middle]:
            right = middle
        else:
            left = middle + 1
    
    if target == list[right]:
        return right
    return None
            

arr = [2, 4, 7, 12, 15, 21, 34, 34, 46, 57, 70, 82, 86, 92, 99]

print(binary_search(arr, 11))

File: Problem_Solving/Algorithm/binary_search.cpp
#include <iostream>

using namespace std;

int binary_search(int *arr, int length, int target){
    int left = 0, right = length - 1;

    while(left < right){
        int middle = (left + right) / 2;

        if(target <= arr[middle]){
            right = middle;
        } else{
            left = middle + 1;
        }
    }

    if(target == arr[right]) return right;
    return -1;
}

int main(){

    int arr[10] = {2,4,7,12,15,21,34,35,46,57};
    int position = binary_search(arr, 10, 4);

    cout << position << endl;
    return 0;
}